apps/src/clientApi.js

```diff
@@ -32,13 +32,25 @@ class CollectionsApi {
     this.collectionType = collectionType;
   }
 
+  withProjectId(projectId) {
+    // TODO: use `new super()` instead of CollectionsApi once we drop
+    // support for ie10.
+    var boundApi = new CollectionsApi(this.collectionType);
+    boundApi.projectId = projectId;
+    return boundApi;
+  }
+
   basePath(path) {
-    return apiPath(this.collectionType, window.dashboard.project.getCurrentId(), path);
+    return apiPath(
+      this.collectionType,
+      this.projectId || window.dashboard.project.getCurrentId(),
+      path
+    );
   }
 
   ajax(method, file, success, error, data) {
     error = error || function () {};
-    if (!window.dashboard) {
+    if (!window.dashboard && !this.projectId) {
```

Comment:
How you have it is fine. I misunderstood what was happening, and thought the logic was wrong (it was me who was wrong).